main.cpp

```diff
@@ -592,6 +619,17 @@ float Preamp::distortSignal(float level, bool allowClip)
     return level;
 }
 
+float Preamp::cutMud(float level)
+{
+    Equalizer eq;
+    while (eq.numBands >= 2 && level > 5)
+    {
+        level -= 3.0f;  
+        return level;
+    }
```

Comment:
this `return level;` means your loop only runs once. 
effectively you have this:
```
if( eq.numBands >= 2 && level > 5 )
{
    level -= 3.0f;
    return level;
}
```
make your function **actually** loop
- [ ] revise